In [1]:
import pandas as pd
import time
import seaborn as sns
import datetime as dt
import numpy as np
import matplotlib.pyplot as plt
import os
from tqdm import tqdm

import pokerDataParser

In [2]:
path = os.path.join('C:\\Users','templ','Documents','GitHub','gambling','texasHoldEm','game_data','Export Holdem Manager 2.0 12302016144830.txt')

t = pokerDataParser.parseGames(path)

test = pokerDataParser.analyzeGame(t[2])

In [3]:
%%timeit
pokerDataParser.analyzeGame(t[2])

1.07 s ± 82.6 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)


In [4]:
print('number of individual games in data: {}'.format(len(t)))

number of individual games in data: 42688


Functions above show the individual computation time to parse the dataset and the total number of games, 42688. 

In [5]:
master = pd.DataFrame()

for data in tqdm(t):
    out = pokerDataParser.analyzeGame(data)
    master = master.append(out)

  0%|                                                                              | 1/42688 [00:00<3:32:53,  3.34it/s]


ValueError: operands could not be broadcast together with shapes (2,) (4,) 

In [ ]:
master